### ISSUE TITLE
UnicodeDecodeError on Import

### ISSUE THREAD

**Issue by Crelam** [AcidWeb/CurseBreaker#335]
I do not have an Wago Addons API key set, however when I use the import function I receive an error I would not expect.

`UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfc in position 239: invalid start byte`

I would have expected to receive the "Wago Addons API key is Missing" error as seen in issue #334.

**Comment by AcidWeb** [comment]
Could you add `CURSEBREAKER_DEBUG` environmental variable? It should make error more verbose.

**Comment by Crelam** [comment]
`CB> import
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Program Files (x86)\World of Warcraft\_retail_\CurseBreaker.py:205 in start                   │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\CurseBreaker.py'                                                             │
│                                                                                                  │
│ C:\Program Files (x86)\World of Warcraft\_retail_\CurseBreaker.py:860 in c_import                │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\CurseBreaker.py'                                                             │
│                                                                                                  │
│ C:\Program Files (x86)\World of Warcraft\_retail_\CB\Core.py:610 in detect_addons                │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\CB\\Core.py'                                                                 │
│                                                                                                  │
│ C:\Program Files (x86)\World of Warcraft\_retail_\CB\Core.py:674 in __init__                     │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\CB\\Core.py'                                                                 │
│                                                                                                  │
│ C:\Program Files (x86)\World of Warcraft\_retail_\CB\Core.py:702 in parse                        │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\CB\\Core.py'                                                                 │
│                                                                                                  │
│ C:\Program Files (x86)\World of Warcraft\_retail_\CB\Core.py:683 in parse_file                   │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\CB\\Core.py'                                                                 │
│                                                                                                  │
│ C:\Program Files (x86)\World of Warcraft\_retail_\codecs.py:322 in decode                        │
│                                                                                                  │
│ [Errno 2] No such file or directory: 'C:\\Program Files (x86)\\World of                          │
│ Warcraft\\_retail_\\codecs.py'                                                                   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfc in position 239: invalid start byte`

**Comment by AcidWeb** [comment]
It looks like one of your addons have something strange within its structure. Could you pack and share entire `Interface\AddOns` directory?

**Comment by Crelam** [comment]
My AddOns directory is rather large given that its also where I store various texture and addon experiments.  I'm embarrassed to say its over 4 GB.  It also contains some files I wouldn't feel comfortable sharing due to copyright or licensing issues.

Is there something in particular I should look into?  Could it be failing on one of my custom addons?  Could the size of the directory itself be the issue?  I also have some changes above the addons folder to use modified icons and textures.

If there is nothing in particular, later today I can just put the addons in one at a time and see when the error occurs to narrow it down.

**Comment by Crelam** [comment]
Would a zip of just the directory structure and the xml and toc files work?  I think I can share them without running afoul of any licensing issues.

**Comment by AcidWeb** [comment]
Nah. I rather check entire structure as it is possible that something else than xml/toc is causing that.

**Comment by Crelam** [comment]
Understood.  I'll still parse though the addons and see if I can narrow it down to a single (or at least the first) problem addon and let you know what I find.

**Comment by AcidWeb** [comment]
Do you have Masque_Shadow?

**Comment by AcidWeb** [comment]
OK. Got the cause I will fix it soon.

**Comment by Crelam** [comment]
> Do you have Masque_Shadow?

Negative.



> OK. Got the cause I will fix it soon.

Excellent!
